code/modules/antagonists/vampire/vampire_powers/gargantua_powers.dm

```diff
@@ -201,3 +203,88 @@
 		var/mob/living/L = user
 		L.apply_status_effect(STATUS_EFFECT_CHARGING)
 		L.throw_at(target, targeting.range, 1, L, FALSE, callback = CALLBACK(L, TYPE_PROC_REF(/mob/living, remove_status_effect), STATUS_EFFECT_CHARGING))
+
+/obj/effect/proc_holder/spell/vampire/arena
+	name = "Desecrated Duel (150)"
+	desc = "You jump towards someone. Upon landing, you conjure an arena, and within it you will heal brute and burn damage, recover from fatigue faster, and be strengthened against lasting damages. Can be recasted to end the spell early."
+	gain_desc = "You can now jump to a target, creating an arena where you land."
+	required_blood = 150
+	base_cooldown = 30 SECONDS
+	action_icon_state = "blood_barrier"
+	should_recharge_after_cast = FALSE
+	/// Is our spell active?
+	var/spell_active = FALSE
+	/// Holds a reference to the timer until either the spell runs out or we recast it
+	var/timer
+	/// Holds a reference to all arena walls so we can qdel them easily with dispell()
+	var/list/all_temp_walls = list()
+
+/obj/effect/proc_holder/spell/vampire/arena/create_new_targeting()
+	var/datum/spell_targeting/click/T = new
+	T.click_radius = 0
+	T.allowed_type = /mob/living/carbon/human
+	return T
+
+/obj/effect/proc_holder/spell/vampire/arena/cast(list/targets, mob/living/user)
+	if(!targets)
+		return
+
+	if(timer) // Recast to dispell the wall and buff early
+		dispell(user)
+		return
+
+	// First we leap towards the enemy target
+	user.add_stun_absorption("gargantua", INFINITY, 2) // We temporarily make the gargantua immune to stuns to stop any matrix fuckery from happening
+	animate(user, 1 SECONDS, pixel_z = 64, flags = ANIMATION_RELATIVE, easing = SINE_EASING|EASE_OUT)
+	addtimer(CALLBACK(user, user.spin(12, 1), 3, 2), 0.3 SECONDS)
+	var/angle = get_angle(user, targets[1]) + 180
```

Comment:
Can you pull `targets[1]` out into a variable so it's clearer what you're referring to